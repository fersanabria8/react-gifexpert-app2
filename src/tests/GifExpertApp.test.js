import React from 'react';
import { shallow } from 'enzyme';
import { GifExpertApp } from "../GifExpertApp"
import '@testing-library/jest-dom'

describe('Pruebas con <GifExpertApp/>', () => {

  test('Debe de tener un snapshot del componente', () => {

    const wrapper = shallow( <GifExpertApp/> )
    expect( wrapper ).toMatchSnapshot();

  });

  test('Debe de mostrar una lista de categorias', () => {


    const categories = ['One Punch', 'Dragon Ball Z'];
    const wrapper = shallow(<GifExpertApp defaultCategories={ categories } />);

    expect( wrapper ).toMatchSnapshot();
    expect( wrapper.find('GifGrid').length ).toBe( categories.length )

  })


});