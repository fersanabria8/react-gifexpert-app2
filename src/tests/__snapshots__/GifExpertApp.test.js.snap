// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con <GifExpertApp/> Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball Z"
      key="Dragon Ball Z"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas con <GifExpertApp/> Debe de tener un snapshot del componente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
